options/ansi/generic/threads.cpp

```diff
@@ -0,0 +1,13 @@
+#include <abi-bits/errno.h>
+#include <mlibc/threads.hpp>
+#include <threads.h>
+
+int thrd_create(thrd_t *thr, thrd_start_t func, void *arg) {
+	int res = mlibc::thread_create(thr, 0, (void * (*)(void *)) func, arg);
```

Comment:
I think casting this and then calling the result is UB, and I think it could also cause problems in cases where `sizeof(int) != sizeof(void*)`. instead I think you could add two arguments corresponding to each of the possible function pointers, or perhaps a union or some such thing. regardless I think this cast should be avoided